src/storage/v2/replication/replication_client.cpp

```diff
@@ -205,68 +214,72 @@ void ReplicationStorageClient::StartTransactionReplication(const uint64_t curren
       // This is a signal to any async streams that are still finalizing to start recovery, since this commit will be
       // missed.
       *locked_state = RECOVERY;
-      return;
+      return replica_stream;
     case MAYBE_BEHIND:
       spdlog::error(
           utils::MessageWithLink("Couldn't replicate data to {}.", client_.name_, "https://memgr.ph/replication"));
       TryCheckReplicaStateAsync(storage, std::move(db_acc));
-      return;
+      return replica_stream;
     case DIVERGED_FROM_MAIN:
       spdlog::error(utils::MessageWithLink("Couldn't replicate data to {} since replica has diverged from main.",
                                            client_.name_, "https://memgr.ph/replication"));
-      return;
+      return replica_stream;
     case READY:
-      MG_ASSERT(!replica_stream_);
       try {
-        replica_stream_.emplace(storage, client_.rpc_client_, current_wal_seq_num, main_uuid_);
+        replica_stream.emplace(storage, client_.rpc_client_, current_wal_seq_num, main_uuid_);
```

Comment:
Move `std::optional<ReplicaStream> replica_stream;` closer by adding a scope for this case ie.
```cpp
case READY: {
  auto replica_stream = std::optional<ReplicaStream>{};
  //...
  return replica_stream;
}
```